roles/openshift_health_checker/openshift_checks/logging/logging.py

```diff
@@ -49,9 +49,10 @@ def get_pods_for_component(self, execute_module, namespace, logging_component, t
 
         return pods['items'], None
 
-    @staticmethod
-    def not_running_pods(pods):
-        """Returns: list of pods not in a ready and running state"""
+    def not_running_pods(self, pods):
+        """Returns: list of pods not in a ready and running state
+        or an OpenShiftCheckException if no pods have been scheduled."""
+        self.ensure_scheduled_pods(pods)
         return [
             pod for pod in pods
             if any(
```

Comment:
We could just stick to the goal of filtering out pods that aren't actually contributing, and fix the check breakage like so:

```
if not pod.get('status', {}).get('containerStatuses') or any(
... and later ...
    for condition in pod['status'].get('conditions', [])
```